src/main/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprinter.java

```diff
@@ -145,30 +109,24 @@ public FragmentFingerprinter(List<String> aFragmentList) throws NullPointerExcep
      *
      * @param aListOfUniqueSmiles is a list that stores fragments in the form of unique SMILES.
      * To be able to calculate the fingerprint for a molecule, the fragments should belong to one molecule.
-     * @return BitSet. BitSet is a CDK class that implements the IBitFingerprint interface of CDK.
+     * @return BitSetFingerprint. BitSetFingerprint is a CDK class that implements the IBitFingerprint interface of CDK.
      * This allows methods to be used that return useful information from the calculated bit fingerprint,
      * such as the number of positive bits in the fingerprint, etc.
-     * @throws NullPointerException is thrown if the list aListOfUniqueSmiles is null.
-     * @throws IllegalArgumentException is thrown if the list aListOfUniqueSmiles contains blank/empty strings.
+     * @throws NullPointerException is thrown if the list param (or any of its elements) is null.
+     * @throws IllegalArgumentException is thrown if the list param contains blank/empty strings.
      */
     @Override
     public IBitFingerprint getBitFingerprint(List<String> aListOfUniqueSmiles) throws NullPointerException, IllegalArgumentException {
-        this.validityCheckOfParameterList(aListOfUniqueSmiles,"aListOfUniqueSmiles (list of string instances) is null.",
-                "aListOfUniqueSmiles (at least one list element) is null.",
-                "aListOfUniqueSmiles (at least one list element) is blank/empty.");
+        this.validityCheckOfParameterList(aListOfUniqueSmiles,"Given list of string instances is null.",
+                "Given list includes at least one null element.",
+                "Given list includes at least one blank/empty element.");
         BitSet tmpBitSet = new BitSet(this.uniqueSmilesToPositionMap.size());
```

Comment:
I don't know whether HashMap cashes it internally but potentially, each call to size() might trigger a complete traversal of the map. To avoid this and because the size of the map does not change in our use case here, I suggest to calculate the size of the map / the fingerprint once in the constructor and store it.